pkg/registry/registry.go

```diff
@@ -115,3 +221,61 @@ func resolveInRegistry(registry *url.URL, ref *url.URL) (rc io.ReadCloser, ok bo
 	}
 	return resp.Body, true, nil
 }
+
+func sortedList(list []json.RawMessage, cursor int, sortBy string, limit int, reverse, starved bool) ([]json.RawMessage, bool) {
+	if cursor-len(list) < limit {
+		if !starved {
+			return list, false
+		}
+		if cursor > len(list) {
+			cursor = len(list)
+		}
+		list = list[cursor:]
```

Comment:
I don't understand how you can remove the beginning of the list before having sorted it. What is the trick?